packages/etherscan/src/lib/etherscan.ts

```diff
@@ -149,117 +241,283 @@ async function getSourceCode(call: Etherscan, address: string) {
+  }))
 }
 
-function dailyUncleBlkCount(call: Etherscan, startDate: Date, endDate: Date, params: Optional<DailyUncleBlockCountRequest>) {
-  return call<DailyUncleBlockCountReponse[]>({
+async function dailyUncleBlkCount(call: Etherscan, startDate: Date, endDate: Date, params: Optional<DailyUncleBlockCountRequest>) {
+  const res = await call<DailyUncleBlockCountReponse[]>({
     module: 'stats',
     action: 'dailyuncleblkcount',
     startdate : formatDate(startDate),
     enddate: formatDate(endDate),
     ...params
   });
+  return res.map(data => ({
+    UTCDate: toDate(data.UTCDate),
+    unixTimeStamp: toNumber(data.unixTimeStamp),
+    uncleBlockCount: toNumber(data.uncleBlockCount),
+    uncleBlockRewards_Eth: toBigNumber(data.uncleBlockRewards_Eth)
+  }))
 }
 
 //////////
 // LOGS //
 //////////
 
-function getLogs(call: Etherscan, address: string, params: Optional<LogsRequest>) {
-  return call<LogsResponse[]>({ module: 'logs', action: 'getLogs', address, ...params });
+async function getLogs(call: Etherscan, address: string, params: Optional<LogsRequest>) {
+  const res = await call<LogsResponse[]>({ module: 'logs', action: 'getLogs', address, ...params });
+  return res.map(data => ({
+    ...data,
+    blockNumber: hexToNumber(data.blockNumber),
+    timeStamp: unixToDate(data.timeStamp),
+    gasPrice: hexToNumber(data.gasPrice),
+    gasUsed: hexToNumber(data.gasUsed),
+    logIndex: hexToNumber(data.logIndex),
+    transactionIndex: hexToNumber(data.transactionIndex),
+  }))
 }
 
 ///////////
 // PROXY //
 ///////////
 
-function ethBlockNumber(call: Etherscan) {
-  return call<string>({ module: 'proxy', action: 'eth_getBlockByNumber' })
+async function blockNumber(call: Etherscan) {
+  return await call<string>({ module: 'proxy', action: 'eth_blockNumber' });
 }
 
-function ethGetBlockByNumber(call: Etherscan, tag: ProxyTag, boolean: boolean) {
+async function getBlockByNumber(call: Etherscan, tag: ProxyTag, boolean: boolean) {
   if (boolean === true) {
-    return call<BlockResponse<TransactionResponse[]>>({ module: 'proxy', action: 'eth_getBlockByNumber', tag, boolean });
+    const res = await call<BlockResponse<TransactionResponse[]>>({ module: 'proxy', action: 'eth_getBlockByNumber', tag, boolean });
+    return{
+      ...res,
+      difficulty: hexToNumber(res.difficulty),
+      baseFeePerGas: toBigNumber(res.baseFeePerGas),
+      gasLimit: toBigNumber(res.gasLimit),
+      gasUsed: toBigNumber(res.gasUsed),
+      nonce: hexToNumber(res.nonce),
+      number: hexToNumber(res.number),
+      size: hexToNumber(res.size),
+      timestamp: unixToDate(res.timestamp),
+      totalDifficulty: hexToNumber(res.totalDifficulty),
+    }
   } else {
-    return call<BlockResponse<string[]>>({ module: 'proxy', action: 'eth_getBlockByNumber', tag, boolean });
+    const res = await call<BlockResponse<string[]>>({ module: 'proxy', action: 'eth_getBlockByNumber', tag, boolean });
+    return{
```

Comment:
Hmmm if the response is a `string[]` why do you return an object ?

Reply:
because I don't return a string[] but a BlockResponse

Reply:
ah my bad I read too quickly